src/api/java/io/github/cvc5/IOracle.java

```diff
@@ -0,0 +1,20 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Mudathir Mohamed
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2022 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * The cvc5 java API.
+ */
+
+package io.github.cvc5;
+
+public interface IOracle {
```

Comment:
If technically possible, I would like the APIs to be consistant. We should either use the same interface in all 3 APIs or use `std::function` equivalents in the Java and Python bindings. The Java equivalent to `std::function` is [`Function<T, R>`](https://docs.oracle.com/javase/8/docs/api/java/util/function/Function.html). I don't think Python has an equivalent, so an interface may be better.

Reply:
We can't use `Function<T, R>` here because the user is expected to call solver functions that may throw a CVC5ApiException.

Reply:
Umm... How is that a problem? Isn't it the user's responsibility to deal with those exceptions? The internals do not expect an exception to be thrown when the oracle function is called...

Reply:
Yes, the user should handle the api exceptions. But I don't need to force the user to surround the body of each Oracle with try-catch block. 
For example, if the Functional<T, R> interface is used, the user needs to change replace this expression  
https://github.com/cvc5/cvc5/pull/8815/files#diff-0f2f1183d810770d1917a0457a53e85ddf5cb52c593c50798d8b63ac1ef6cc66R3020
with try catch block. 
Otherwise compile error would be thrown.